aspnetcore/mvc/views/tag-helpers/builtin/resources/CacheTagHelper.md

```diff
@@ -0,0 +1,286 @@
+﻿[Back To Built In Tag Helpers List](../../builtin.md)
+
+# Cache Tag Helper
+
+By [Peter Kellner](http://peterkellner.net) 
+
+
+The  Cache Tag Helper provides the ability to dramatically improve the performance of your ASP.NET Core app by caching its content to the internal ASP.NET Core cache provider.
+
+A simple example that shows the Cache Tag Helper in action sets the current time in its content area. The Razor View Engine sets the default `expires-after` to twenty minutes, which is the default if you specify no additional attributes.
+
+Here is an example of the code you include in your `cshtml` page to achieve the default cache behavior described above.
+
+```html
+<Cache>@DateTime.Now<Cache>
+```
+
+If you repeatedly refresh the page that contains `CacheTagHelper`, the first time the current`DateTime`will be shown. Repeated refreshes of the page will continue to show the first `DateTime` shown until the cache expires (which is likely to be twenty minutes unless server memory pressure evicts the cache sooner).
+
+You can achieve more control of the cache duration of the content by setting any of the following attributes:
+
+## Cache Tag Helper Attributes
+
+- - -
+
+### enabled    
+
+
+| Attribute Type 	| Valid Values   	|
+|----------------	|----------------	|
+| boolean         	| "true" (default) 	|
+|               	| "false"	|
+
+
+The attribute `enabled` determines whether any caching is performed of the content inside the Cache Tag Helper. The default is `true`, which results in all other attributes being valid.  The default value is `true`.  If set to `false`, regardless of what other attributes are specified, this Cache Tag Helper will have no caching effect on the rendered output.
+Usage Example:
+
+```html
+<Cache enabled="true">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+- - -
+
+### expires-on 
+
+| Attribute Type 	| Example Value   	|
+|----------------	|----------------	|
+| DateTimeOffset    | "@new DateTime(2025,1,29,17,02,0)" 	|
+
+
+The attribute `expires-on` sets a deterministic expiration date for caching the content of the Cache Tag Helper when it is first rendered. The example below will cache the contents of the Cache Tag Helper until 5:02 PM on January 29, 2025.
+
+Usage Example:
+
+```html
+<Cache expires-on="@new DateTime(2025,1,29,17,02,0)">
+    Current Time Inside Cache Tag Helper: @DateTime.Now
+</Cache>
+```
+
+- - -
+
+### expires-after
+
+| Attribute Type 	| Example Value   	|
+|----------------	|----------------	|
+| TimeSpan    | "@TimeSpan.FromSeconds(120)" 	|
+
+
+The attribute `expires-after` sets a specific amount of time from the current `DateTime` to cache the contents of the Cache Tag Helper. This value is set when it is first rendered.  The example below caches the contents of the Cache Tag Helper for one hundred twenty seconds. If there is memory pressure on the server, the cache may be evicted sooner.
```

Comment:
* 2 spaces after first sentence
* "one hundred twenty seconds" ... to "120 seconds"
* "the cache may be evicted sooner" ... mmm ... "cached entities" or "cached items" or something like that. It's not really the whole cache necessarily.
* You can drop both state verbs if you want ...

If memory pressure develops on the server, the app can evict cached items to free memory.